### PR TITLE
checking instance type

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: liuwei <liuweixa@redhat.com>

### CODE REVIEW

**Comment by skeeey** [comment]
refer to https://github.com/open-cluster-management/backlog/issues/8802

File: pkg/cloud/aws/aws.go [hunk 1]
```diff
@@ -331,7 +333,32 @@ func (a *awsProvider) findSubnet(vpcId string) (*ec2.Subnet, error) {
...
+					Values: []*string{aws.String(instanceType)},
+				},
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// the instance type can be supported in this subnet
+		if len(output.InstanceTypeOfferings) == 1 {
```
Comment (qiujian16) [inline pkg/cloud/aws/aws.go:356]: what does 1 mean here.

File: pkg/cloud/aws/aws.go [hunk 2]
```diff
@@ -331,7 +333,32 @@ func (a *awsProvider) findSubnet(vpcId string) (*ec2.Subnet, error) {
 			},
 		}
 	}
-	return subnets.Subnets[0], nil
+
+	for _, subnet := range subnets.Subnets {
+		output, err := a.awsClinet.DescribeInstanceTypeOfferings(&ec2.DescribeInstanceTypeOfferingsInput{
```
Comment (qiujian16) [inline pkg/cloud/aws/aws.go:338]: it needs doc to describe the intention of this  call

File: pkg/cloud/aws/aws.go [hunk 3]
```diff
@@ -482,7 +509,7 @@ func (a *awsProvider) tagSubnet(vpc *ec2.Vpc) (*ec2.Subnet, error) {
 		return nil, err
 	}
 
-	// the tag has been labeled
+	// the subnet has been labeled with internal ELB label, just return it
```
Comment (qiujian16) [inline pkg/cloud/aws/aws.go:512]: we need some reason, more doc please

File: pkg/cloud/aws/aws.go [hunk 4]
```diff
@@ -508,26 +539,57 @@ func (a *awsProvider) tagSubnet(vpc *ec2.Vpc) (*ec2.Subnet, error) {
 }
 
 func (a *awsProvider) untagSubnet(vpc *ec2.Vpc) error {
-	subnet, err := a.findSubnet(*vpc.VpcId)
-	if errors.IsNotFound(err) {
-		return nil
-	}
+	// only untag the subnet that was tagged by addon
```
Comment (qiujian16) [inline pkg/cloud/aws/aws.go:542]: reason?

File: pkg/cloud/aws/aws.go (hunk 1 above)
Comment (skeeey) [reply pkg/cloud/aws/aws.go]: add comments in the code

File: pkg/cloud/aws/aws.go (hunk 2 above)
Comment (skeeey) [reply pkg/cloud/aws/aws.go:339]: add comments in the code

File: pkg/cloud/aws/aws.go (hunk 3 above)
Comment (skeeey) [reply pkg/cloud/aws/aws.go]: add comments in the code

File: pkg/cloud/aws/aws.go (hunk 4 above)
Comment (skeeey) [reply pkg/cloud/aws/aws.go]: add comments in the code

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**